capsules/extra/src/log.rs

```diff
@@ -824,7 +824,7 @@ impl<'a, F: Flash + 'static> flash::Client<F> for Log<'a, F> {
                     _ => unreachable!(),
                 }
             }
-            flash::Error::FlashError | flash::Error::FlashMemoryProtectionError => {
+            Err(_) => {
```

Comment:
It would be nice to not change that these are specified (and future errors would have to be added explicitly).

Reply:
It looks like the code inside the block doesn't care if it is `FlashError`, `FlashMemoryProtectionError`, nor any other error. It matters only if it's an error or not. Explicitly matching each error code might just annoy the programmer without any benefit for the code's logic. Am I wrong?

Reply:
Well I'm not sure that the original author didn't specifically choose to not use the catchall. It's easier for me to approve this PR if the change is just mechanistic, and not actually making different design decisions.

Reply:
Matched errors explicitly.